src/components/common/YourCompany/YourCompany.vue

```diff
@@ -502,8 +510,9 @@ export default class YourCompany extends Mixins(
 
   /** Reset company name values to original. */
   protected resetName () {
-    // reset business information
-    this.setBusinessInformation(this.getEntitySnapshot.businessInfo)
+    // reset business information, except for association type.
+    const businessInfo = { ...this.getEntitySnapshot.businessInfo, associationType: this.getAssociationType }
+    this.setBusinessInformation(businessInfo)
 
```

Comment:
When you reset name, do you also need to reset the filingData array?

Reply:
seems to be handled by the watch on `businessNameChanged`